include/rcpputils/filesystem_helper.hpp

```diff
@@ -547,28 +547,32 @@ inline bool remove(const path & p)
  */
 inline bool remove_all(const path & p)
 {
-  if (!is_directory(p)) {return false;}
+  if (!is_directory(p)) {return remove(p);}
 
 #ifdef _WIN32
   // We need a string of type PCZZTSTR, which is a double null terminated char ptr
   size_t length = p.string().size();
   TCHAR * temp_dir = new TCHAR[length + 2];
   memcpy(temp_dir, p.string().c_str(), length);
-  temp_dir[length] = 0;
-  temp_dir[length + 1] = 0;    // double null terminated
+  temp_dir[length] = '\0';
+  temp_dir[length + 1] = '\0';  // double null terminated
 
   SHFILEOPSTRUCT file_options;
   file_options.hwnd = nullptr;
-  file_options.wFunc = FO_DELETE;    // delete (recursively)
+  file_options.wFunc = FO_DELETE;  // delete (recursively)
   file_options.pFrom = temp_dir;
   file_options.pTo = nullptr;
-  file_options.fFlags = FOF_NOCONFIRMATION | FOF_SILENT;    // do not prompt user
+  file_options.fFlags = FOF_NOCONFIRMATION | FOF_SILENT;  // do not prompt user
   file_options.fAnyOperationsAborted = FALSE;
   file_options.lpszProgressTitle = nullptr;
   file_options.hNameMappings = nullptr;
 
-  SHFileOperation(&file_options);
+  auto ret = SHFileOperation(&file_options);
   delete[] temp_dir;
+
+  if (0 != ret) {
+    return false;
+  }
```

Comment:
```suggestion
  return 0 == ret && FALSE == file_options.fAnyOperationsAborted;
```

> It is good practice to examine the value of the fAnyOperationsAborted member of the SHFILEOPSTRUCT. SHFileOperation can return 0 for success if the user cancels the operation. If you do not check fAnyOperationsAborted as well as the return value, you cannot know that the function accomplished the full task you asked of it and you might proceed under incorrect assumptions.

Based on the docs, I would check `fAnyOperationsAborted` (though the user can't cancel the operation in our case).